src/Applications/UUPMediaCreator/Pages/DownloadPage.xaml.cs

```diff
@@ -44,7 +45,10 @@ private async void DownloadPage_Loaded(object sender, RoutedEventArgs e)
 
             App.ConversionPlan.TmpOutputFolder = await UpdateUtils.ProcessUpdateAsync(App.ConversionPlan.UpdateData, tmp.Path, App.ConversionPlan.MachineType, this, App.ConversionPlan.Language, App.ConversionPlan.Edition, UseAutomaticDownloadFolder: false).ConfigureAwait(false);
 
-            Frame.Navigate(typeof(BuildingISOPage));
+            await Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () => 
```

Comment:
Is this needed here? This code in the UI thread, the code that doesn't is in the report function indeed. I would keep the dispatcher for the report function but I would remove it here.

Reply:
Yeah, for some reason it threw a wrong thread exception, so I added that in.